### PR TITLE
Consume go-state-types

### PR DESCRIPTION
**Pull request by ZenGround0**
This is blocked on specs actors consuming go-state-types on release branch 9 at upcoming tag v0.9.4 

FYI @anorth @Stebalien

### CODE REVIEW

File: proofs.go [hunk 1]
```diff
@@ -131,7 +132,7 @@ func VerifyWindowPoSt(info abi.WindowPoStVerifyInfo) (bool, error) {
 
 // GeneratePieceCommitment produces a piece commitment for the provided data
 // stored at a given path.
-func GeneratePieceCID(proofType abi.RegisteredSealProof, piecePath string, pieceSize abi.UnpaddedPieceSize) (cid.Cid, error) {
+func GeneratePieceCID(proofType aabi.RegisteredSealProof, piecePath string, pieceSize abi.UnpaddedPieceSize) (cid.Cid, error) {
```
Comment (Stebalien) [inline proofs.go:135]:
```suggestion
func GeneratePieceCID(proofType aabi.RegisteredSealProof, piecePath string, pieceSize aabi.UnpaddedPieceSize) (cid.Cid, error) {
```

Comment (ZenGround0) [reply proofs.go:135]: ah these should both be `abi`

**Comment by anorth** [comment]
I recommend waiting for https://github.com/filecoin-project/specs-actors/pull/1096

**Comment by anorth** [comment]
These `SealVerifyInfo` and friends are currently in specs-actors, not go-state-types, so there will be two versions of them. Presumably we don't want to version this ffi package with release branches, though (it will need to add new functions if things change). We'll alias on to the types from the 0.9 branch. Until we do so, though, we might need  a release branch for testnet.

**Comment by anorth** [comment]
https://github.com/filecoin-project/specs-actors/pull/1097

**Review by Stebalien — APPROVED** [review]
LGTM but I have _no_ idea how we'd switch between proof types at runtime. For the next few network upgrades, we'll probably be fine here, but this could be a bit tricky.

**Comment by ZenGround0** [comment]
Yes we should definitely put this on the list of things we need to figure out for future upgrades.

**Pull request merged**